app/grandchallenge/archives/views.py

```diff
@@ -732,3 +737,20 @@ def studies(self, request, pk=None):
             .distinct("study_description")
         )
         return Response(studies)
+
+
+class ArchiveItemViewSet(ReadOnlyModelViewSet):
+    serializer_class = ArchiveItemSerializer
+    permission_classes = [ArchiveItemPermission]
+    filter_backends = [DjangoFilterBackend]
+    filterset_fields = ["archive"]
+
+    def get_queryset(self):
+        # restrict queryset to items from archives the user has view permission for
```

Comment:
See the comment above, it seems the permissions are being done in two places that need maintaining.

Reply:
Yes, there are two checks, one for the object itself (to retrieve a single archive item, normally done with `DjangoObjectPermissions`) and one for retrieving a list of archive items (which is normally done through a filter `ObjectPermissionsFilter` which filters the list to those objects the user has view permission for). Again, since we need to check them for the archive and not the item, that filter does not work for us. All the filter does is filter the queryset, so I thought it would be easiest to just override the get_queryset method, but I can also create a custom filter class.